sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/SynchronizedAccessor.java

```diff
@@ -38,12 +38,11 @@ public SynchronizedAccessor(Supplier<T> supplier) {
                 return Mono.just(cache);
             }
             if (!wip.getAndSet(true)) {
-                try {
-                    cache = supplier.get();
-                    sink.next(cache);
-                } catch (Exception e) {
-                    sink.error(e);
-                }
+                return supplier.get()
+                    .doOnNext(val -> cache = val)
+                    .doOnNext(sink::next)
+                    .doOnError(sink::error)
```

Comment:
Do not use `doOn*` methods. They are for side-effects operations only.

Reply:
doOnNext will get triggered when the data gets emitted successfully. 
what can be the issue here?